src/coreclr/src/jit/regalloc.cpp

```diff
@@ -175,6 +175,7 @@ regNumber Compiler::raUpdateRegStateForArg(RegState* regState, LclVarDsc* argDsc
         }
         else
         {
+            assert(!regState->rsIsFloat);
```

Comment:
We could miss this check due to the `break` below, then fail later in an unexpected way.

Reply:
cc: @dotnet/jit-contrib